### ISSUE TITLE
Get 403 error when query GitHub HTTP API on macOS

### ISSUE THREAD

**Issue by lzutao** [kennytm/rustup-toolchain-install-master#28]
```
fetching master commit hash... 

error: https://api.github.com/repos/rust-lang/rust/commits/master: Client Error: 403 Forbidden
```

See this CI line: https://travis-ci.com/kennytm/rustup-toolchain-install-master/jobs/245113302#L444

**Comment by phansch** [comment]
Could this be, because reqwest had a patch version bump [here](https://github.com/kennytm/rustup-toolchain-install-master/pull/26/files#diff-d2ede298d9a75c849170d6ef285eda1eR959)?

**Comment by phansch** [comment]
A wild guess: Assuming we didn't do anything wrong, I think this *may* be caused by https://github.com/seanmonstar/reqwest/commit/5b55aee1a9ddf785f82d9086c8befc50db268cb8, added in 0.9.21. When GitHub doesn't find a user-agent header in the request, it [will return `403 Forbidden`](https://developer.github.com/v3/#user-agent-required).

**Comment by lzutao** [comment]
So the work around is to rollback reqwest ?

**Comment by lzutao** [comment]
Should be closed by  #27 .

**Comment by lzutao** [comment]
Error with reqwest 0.9.20: https://travis-ci.com/rust-lang/rust-clippy/jobs/245150001#L484-L490

**Comment by lzutao** [comment]
Check it out with curl on macos:
```console
$ curl -v -H 'accept: application/vnd.github.VERSION.sha' https://api.github.com/repos/rust-lang/rust/commits/master
< HTTP/1.1 403 Forbidden
< Server: GitHub.com
< Date: Mon, 14 Oct 2019 04:05:54 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 256
< Status: 403 Forbidden
< X-RateLimit-Limit: 60
< X-RateLimit-Remaining: 0
< X-RateLimit-Reset: 1571029486
< X-GitHub-Media-Type: github.v3; param=VERSION.sha
< Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
< Access-Control-Allow-Origin: *
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< X-Frame-Options: deny
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
< Content-Security-Policy: default-src 'none'
< X-GitHub-Request-Id: 6874:0E4C:41E111:6E4229:5DA3F422
< 
{ [256 bytes data]
{
  "message": "API rate limit exceeded for 199.7.166.17. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

**Comment by lzutao** [comment]
This should by partially fixed by #27 .
